### ISSUE TITLE
Name of command to edit cell

### ISSUE THREAD

**Issue by jonathandturner** [nushell/nushell#62]
Our view of table data is read-only, but as we think about round-tripping with load/save, it would be helpful to be able to edit cells easily

**Comment by jonathandturner** [comment]
Example in practice:

```
| edit package.version "1.3.2"
```

The verb `update` is another possibility

**Comment by DanielRosenwasser** [comment]
I like `update`, but how do you get a handle to the larger structure being updated, as well as the current value?

**Comment by jonathandturner** [comment]
@DanielRosenwasser - There are a couple convenience methods.

To update, you'll need to take the whole structure in via the pipeline, and then have the user pass the field to update (you can see `inc` for an example of doing this)

To get the current value at the given field path (taken from `inc`):

```
value.item.get_data_by_path(value.tag(), field)
```

And to update the value at the given field path:

```
value.item.replace_data_at_path(value.tag(), f, replacement.item.clone())
```

The `replace_data_at_path` will replace the previous data with the new data and return the whole structure.

**Comment by rrichardson** [comment]
what about `put` as a verb for this? to complement `get` ?

**Comment by waldyrious** [comment]
> what about `put` as a verb for this? to complement `get`?

Another option could be `set`.

**Comment by jonathandturner** [comment]
Now that we have `edit` as the feature, we may want to change this to figuring out the name for it.

**Comment by rrichardson** [comment]
When it comes to changing the values within an existing column of a table, "edit" makes sense.  
I think `set` makes sense as well and it's a bit shorter.

**Comment by ubsan** [comment]
I think `update` makes the most sense, as in "functional record update", since one is not actually modifying any fields.

**Comment by wycats** [comment]
I think `update` is pretty nice, and makes sense both in programming terms and in human terms.

**Comment by waldyrious** [comment]
For future reference, `edit` was renamed to `update` in #1724 (released in nu 0.14.0).
